### PR TITLE
Fix break time while submerged in water

### PR DESCRIPTION
**Pull request by davchoo**
This should fix https://github.com/GeyserMC/Geyser/issues/3091
Mostly moving around code for reuse.

### CODE REVIEW

File: core/src/main/java/org/geysermc/geyser/level/physics/CollisionManager.java [hunk 1]
```diff
@@ -405,6 +405,17 @@ public boolean isPlayerInWater() {
         return session.getGeyser().getWorldManager().getBlockAt(session, session.getPlayerEntity().getPosition().toInt()) == BlockStateValues.JAVA_WATER_ID;
     }
 
+    public boolean isWaterInEyes() {
+        double eyeX = playerBoundingBox.getMiddleX();
+        double eyeY = playerBoundingBox.getMiddleY() - playerBoundingBox.getSizeY() / 2d + session.getEyeHeight() - (1 / 9d);
+        double eyeZ = playerBoundingBox.getMiddleZ();
+
+        int blockID = session.getGeyser().getWorldManager().getBlockAt(session, (int) Math.floor(eyeX), (int) Math.floor(eyeY), (int) Math.floor(eyeZ));
```
Comment (Camotoy) [inline core/src/main/java/org/geysermc/geyser/level/physics/CollisionManager.java:413]: Here you can probably use Nukkit's GenericMath.floor instead.

File: core/src/main/java/org/geysermc/geyser/level/physics/CollisionManager.java [hunk 2]
```diff
@@ -405,6 +405,17 @@ public boolean isPlayerInWater() {
         return session.getGeyser().getWorldManager().getBlockAt(session, session.getPlayerEntity().getPosition().toInt()) == BlockStateValues.JAVA_WATER_ID;
     }
 
+    public boolean isWaterInEyes() {
+        double eyeX = playerBoundingBox.getMiddleX();
+        double eyeY = playerBoundingBox.getMiddleY() - playerBoundingBox.getSizeY() / 2d + session.getEyeHeight() - (1 / 9d);
```
Comment (Camotoy) [inline core/src/main/java/org/geysermc/geyser/level/physics/CollisionManager.java:410]: 9 here seems to be a magic number - can it be elaborated on?

File: core/src/main/java/org/geysermc/geyser/level/block/BlockStateValues.java [hunk 3]
```diff
@@ -458,6 +457,30 @@ public static int getWaterLevel(int state) {
...
+     * @param state BlockState of the block
+     * @return The water height or -1 if the block does not contain water
+     */
+    public static double getWaterHeight(int state) {
+        int waterLevel = BlockStateValues.getWaterLevel(state);
+        if (BlockRegistries.WATERLOGGED.get().contains(state)) {
+            waterLevel = 0;
+        }
+        if (waterLevel >= 0) {
+            double waterHeight = 1 - (waterLevel + 1) / 9.0;
```
Comment (Camotoy) [inline core/src/main/java/org/geysermc/geyser/level/block/BlockStateValues.java:474]: Same deal with the magic number here.

Comment (davchoo) [reply core/src/main/java/org/geysermc/geyser/level/block/BlockStateValues.java]: Both magic numbers refer to the number of levels of water. I've added a constant `NUM_WATER_LEVELS` to address this.

**Review by Camotoy — APPROVED** [review]

**Pull request merged**